python/src/pyripherals/peripherals/I2CController.py

```diff
@@ -163,14 +163,10 @@ def i2c_receive(self, data_length, results='wire'):
                         self.fpga.xem.ActivateTriggerIn(
                             self.endpoints['MEMREAD'].address, self.endpoints['MEMREAD'].bit_index_low)
                     return data
-                if results.lower() == 'pipe':
-                    # should not be used. OK pipe is not configured.
-                    data_read = np.int(np.ceil(data_length*4/16)*16)
-                    buf, e = self.read_pipe_out(0xA0 + 0, data_read)
-                    # reset FIFO
-                    self.fpga.xem.ActivateTriggerIn(
-                        self.endpoints['MEMREAD'].address, self.endpoints['MEMREAD'].bit_index_low)
-                    return list(buf[0::4])[0:data_length]
+                if data_transfer.lower() == 'pipe':
+                    # this does not take care of the FIFO reset
+                    # wait so next data is ready, then continue
+                    time.sleep(0.01)
```

Comment:
It seems to me that if `data_transfer` is set to `'pipe'`, then no data is returned from this function. Why is that?

Reply:
@Ajstros sorry for the confusion! Currently there are separate methods that are being used to read out and clear the pipe (`read_i2c_pipe` and `reset_i2c_fifo`, which are both located in the `TMF8801_sandbox.py` file), the reason for the change in `I2CController.py` is to avoid reading data through the wire in the process of reading it through the pipe, which is why there is only a `time.sleep` call and nothing else. The `time.sleep` is included because without it the pipe is occasionally read before being filled, creating incomplete or corrupted histograms.

Here are the methods I mentioned if you want to take a look! And definitely let me know if you have more questions.
def read_i2c_pipe(tof_dev, data_len=256):
    return tof_dev.fpga.read_pipe_out(tof_dev.endpoints['PIPE_OUT'].address, 
                                      data_len)

def reset_i2c_fifo(tof_dev):
    tof_dev.fpga.xem.ActivateTriggerIn(
            tof_dev.endpoints['FIFO_RESET'].address, tof_dev.endpoints['FIFO_RESET'].bit_index_low)  # High and low bit indexes are the same for the trigger because it is 1 bit wide